### PR TITLE
Speedup BetaGeoModel tests

### PR DESCRIPTION
**Pull request by ricardoV94**
This PR removes the costly fitting from the setup class. Even running a small test like `test_model_repr` would take around one minute as it would have to wait for the useless fitting (in the context of that test) to be done.

Also simplified the patching of the lifetimes reference model. It's enough to override `params_` after initializing it.

Finally added a `pytest.mark.slow` for the `test_model_convergence` so it doesn't run by default (it does run in our CI here). This way the tests take just ~2 seconds on my slow laptop.

### CODE REVIEW

**Comment by ColtAllen** [comment]
To speed this up further, should the model fitting in `setup_class()` be done with MAP? `test_model_convergence()` is already covering a MCMC fit.

File: tests/clv/models/test_beta_geo.py [hunk 1]
```diff
@@ -147,7 +105,7 @@ def test_model_convergence(self, N, rtol):
             recency=recency,
             T=T,
         )
-        model.fit(chains=1, progressbar=False, random_seed=self.rng)
+        model.fit(chains=1, progressbar=False, random_seed=rng)
```
Comment (ColtAllen) [inline tests/clv/models/test_beta_geo.py:108]: Should a MAP fit also be added here for the CI?

File: tests/clv/models/test_beta_geo.py [hunk 2]
```diff
@@ -160,63 +118,133 @@ def test_expected_probability_alive(self):
...
         assert est_prob_alive.dims == ("chain", "draw", "customer_id")
 
         np.testing.assert_allclose(
-            true_prob_alive.mean(),
+            alive.mean(),
             est_prob_alive.mean(),
             rtol=0.05,
         )
 
     def test_expected_num_purchases(self):
```
Comment (ColtAllen) [inline tests/clv/models/test_beta_geo.py:166]: What's the justification for combining these two tests? These are two separate methods with different input and output dimensions.

**Review by ColtAllen — COMMENTED** [review]
`BetaGeoModel` is being redefined in every test. Would it make more sense to define it only once in `setup_class()` as was the case previously? Same with `lifetimes.BetaGeoFitter` as it could be compared against in `test_expected_probability_alive`.  This would also simplify required changes if we decide to implement https://github.com/pymc-labs/pymc-marketing/issues/100.

File: tests/clv/models/test_beta_geo.py [hunk 3]
```diff
@@ -160,63 +118,133 @@ def test_expected_probability_alive(self):
...
-        true_prob_alive = self.alive.mean()  # scalar
-        est_prob_alive = self.fixed_model.expected_probability_alive(
-            self.customer_id,
-            self.frequency,
-            self.recency,
-            self.T,
+        rng = np.random.default_rng(152)
+
+        N = 100
+        # Almost deterministic p = .02
```
Comment (larryshamalama) [inline tests/clv/models/test_beta_geo.py:124]: What does an almost deterministic dropout probability here check?

File: tests/clv/models/test_beta_geo.py [hunk 4]
```diff
@@ -160,63 +118,133 @@ def test_expected_probability_alive(self):
...
+
+        bg_model = BetaGeoModel(
+            customer_id=customer_id,
+            frequency=frequency,
+            recency=recency,
+            T=T,
+            a_prior=pm.Normal.dist(a, 1e-3),
+            b_prior=pm.Normal.dist(b, 1e-3),
+            alpha_prior=pm.Normal.dist(alpha, 1e-3),
+            r_prior=pm.Normal.dist(r, 1e-3),
```
Comment (larryshamalama) [inline tests/clv/models/test_beta_geo.py:141]: Why the informative priors here?

**Review by larryshamalama — COMMENTED** [review]
Looks great, thanks @ricardoV94! The initial solution of having the model fit without the `pytest.mark.slow` was a little naive and due for a speed up. I just added some questions regarding some parameter choices in the tests.

File: tests/clv/models/test_beta_geo.py (hunk 3 above)
Comment (ricardoV94) [reply tests/clv/models/test_beta_geo.py:124]: That the probability of dropout is stable at a value that should induce a nice `p_alive` ~ 0.5, nothing too low nor too high

File: tests/clv/models/test_beta_geo.py (hunk 4 above)
Comment (ricardoV94) [reply tests/clv/models/test_beta_geo.py:141]: For a fake posterior based on prior_predictive sampling around the true values

File: tests/clv/models/test_beta_geo.py (hunk 2 above)
Comment (ricardoV94) [reply tests/clv/models/test_beta_geo.py:166]: They both require building the same type of values, patching the pymc-makerting and lifetimes model, and they are both tested against the same lifetimes method. There was a TODO before suggesting they be merged already, probably for similar reasoning

**Comment by ricardoV94** [comment]
> `BetaGeoModel` is being redefined in every test. Would it make more sense to define it only once in `setup_class()` as was the case previously? Same with `lifetimes.BetaGeoFitter` as it could be compared against in `test_expected_probability_alive`. This would also simplify required changes if we decide to implement #100.

No, I think we should create it every time, because we usually monkey-patch the fit result and we don't want to be operating on the same stateful object (i.e., erasing a previous fit).

It was a conscious choice not to create a class-level model in the other model tests, like the gamma-gamma.

File: tests/clv/models/test_beta_geo.py (hunk 3 above)
Comment (ricardoV94) [reply tests/clv/models/test_beta_geo.py]: Added a comment

File: tests/clv/models/test_beta_geo.py (hunk 4 above)
Comment (ricardoV94) [reply tests/clv/models/test_beta_geo.py]: Replaced by direct sampling with numpy when creating the fake fit

File: tests/clv/models/test_beta_geo.py (hunk 2 above)
Comment (ricardoV94) [reply tests/clv/models/test_beta_geo.py:168]: Done

**Comment by ricardoV94** [comment]
I set this as draft as it includes the commit from #198 

Otherwise I think I addressed your review comments @larryshamalama and @ColtAllen. Let me know if there's anything else.

**Review by larryshamalama — APPROVED** [review]
Looks good to me :)
